AEPCampaignClassic/Sources/CampaignClassicRegistrationManager.swift

```diff
@@ -0,0 +1,144 @@
+/*
+ Copyright 2022 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+import AEPServices
+import AEPCore
+
+///
+/// Class that manages Campaign Classic device registration requests.
+///
+class CampaignClassicRegistrationManager {
+
+    private var runtime: ExtensionRuntime
+    private var systemInfoService: SystemInfoService {
+        ServiceProvider.shared.systemInfoService
+    }
+
+    /// DataStore instance to access Campaign Classic's stored properties.
+    let datastore = NamedCollectionDataStore(name: CampaignClassicConstants.EXTENSION_NAME)
+
+    /// A variable holding the SHA256 hashed device registration information.
+    /// The following variable is a hash of deviceToken, userKey and xml serialized additional parameters provided from the user.
+    /// This property is set if the device is successfully registered with the campaign servers.
+    /// This property is then used to hold back future device registration calls with unchanged data.
+    /// The setter and getter of this property will use the datastore to store and retrieve the information.
+    private(set) var hashedRegistrationData: String? {
+        get {
+            datastore.getString(key: CampaignClassicConstants.DatastoreKeys.TOKEN_HASH)
+        }
+        set {
+            datastore.set(key: CampaignClassicConstants.DatastoreKeys.TOKEN_HASH, value: newValue)
```

Comment:
do we need to consider a nil value in the setter for removing the hash from the datastore entirely?

Reply:
with our code logic today we never set the hash to `nil`
Once we set the registrationHash it's always persisted until next valid hash is to be stored

Reply:
Spoke too soon, we change it to nil. when the privacy switches to opted-out. I have made changes to remove the key if the data is nil.